### PR TITLE
Updated mods helper and class names according to the new BEM standards

### PR DESCRIPTION
**Pull request by AlexDudar**
Updated according to the new BEM standards (see [here](https://en.bem.info/methodology/quick-start/#modifier))

Modified:
- mods helper.js 
- modifiers class names
- unit tests

Modifiers were changed from `block-name--modifier` to `block-name_modifier`

### CODE REVIEW

File: components/button/button.js [hunk 1]
```diff
@@ -16,15 +16,15 @@ function Button({ children, mods = [], size, href, onClick, type, variation, dis
   }
 
   if (size) {
-    mods.push(`size_${size}`);
+    mods.push(`size-${size}`);
```
Comment (asci) [inline components/button/button.js]: According to BEM naming convention the dash `-` could be used only for breaking words in long names, but not for separation values

Comment (AlexDudar) [inline components/button/button.js]: yes, as we discussed yesterday, there is this possibility. I think we can do this, but let's all align, @mAiNiNfEcTiOn @iwwwi what do you think? Actually which one do you prefer `ui-button_size-s` or `ui-button_size_s` ?

Comment (asci) [inline components/button/button.js]: just as an additional example (not in button, but for some future components):
`app-bar_show-icon-left` or `app-bar_show-icon_left`?

**Comment by AlexDudar** [comment]
@asci I made adjustments according to your comment, please check

**Pull request merged**